### ISSUE TITLE
Ref escape analysis errors are not reported for compound assignment 

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#63852]
No error is reported for `x += y` below. Verified the repro with csc.exe 3.11 and latest 4.4 preview.
```csharp
using System;

ref struct S
{
    public static implicit operator S(Span<int> s) => default;
    public static S operator +(S a, S b) => b;
}

class Program
{
    static void F(ref S x)
    {
        Span<int> s = stackalloc int[10];
        S y = s;
        x += y;    // no error
        x = x + y; // error CS8352: 'y' ... may expose referenced variables outside of their declaration scope
    }
}
```
